### PR TITLE
Rethinking variables & constraints dictionaries

### PR DESCRIPTION
**Pull request by a-zakir**

### CODE REVIEW

File: src/solver/optimisation/variables/VariableManagement.h [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        return nativeOptimVar_.NumeroDeVariableDefaillanceNegative[index];
+    }
+
+private:
+    CORRESPONDANCES_DES_VARIABLES& nativeOptimVar_;
+    std::vector<int>& NumeroDeVariableStockFinal;
+    std::vector<std::vector<int>>& NumeroDeVariableDeTrancheDeStock;
+};
+
+class VariableManagerFactory
```
Comment (pet-mit) [inline src/solver/optimisation/variables/VariableManagement.h]: I'm not a big fan of the name of this class. The name "Factory" wrongly suggests that it implements the Factory design pattern. But you only have on VariableManager implementation.

**Review by pet-mit — COMMENTED** [review]
Hello, @flomnes asked me to review this PR.
I only have one small remark. Other than that, it seems to make the code neater overall, so I have no further comments!

File: src/solver/optimisation/variables/VariableManagement.h (hunk 1 above)
Comment (a-zakir) [reply src/solver/optimisation/variables/VariableManagement.h]: Thank you.

Comment (a-zakir) [reply src/solver/optimisation/variables/VariableManagement.h:138]: I need your lights

Comment (a-zakir) [reply src/solver/optimisation/variables/VariableManagement.h:138]: More

Comment (a-zakir) [reply src/solver/optimisation/variables/VariableManagement.h:138]: NOT FOR THIS!

**Comment by flomnes** [comment]
Doxygen doc

File: src/solver/optimisation/adequacy_patch_csr/construct_problem_variables.cpp [hunk 2]
```diff
@@ -106,9 +103,7 @@ void HourlyCSRProblem::constructVariableFlows()
             int algebraicFluxVar;
             int directVar;
             int indirectVar;
-            algebraicFluxVar
-              = CorrespondanceVarNativesVarOptim.NumeroDeVariableDeLInterconnexion[Interco]
-              = NumberOfVariables;
+            algebraicFluxVar = variable_manager.NTCDirect(Interco) = NumberOfVariables;
```
Comment (flomnes) [inline src/solver/optimisation/adequacy_patch_csr/construct_problem_variables.cpp]:
```suggestion
            algebraicFluxVar = variable_manager.NTCDirect(triggeredHour, Interco) = NumberOfVariables;
```

**Comment by flomnes** [comment]
Rename 
- `variable_manager` -> `variableManager`
- `variable_manager_` -> `variableManager`

File: src/solver/optimisation/variables/VariableManagement.cpp [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+ NombreDePasDeTempsPourUneOptimisation_(NombreDePasDeTempsPourUneOptimisation)
+{
+}
+
+int VariableManager::GetShiftedTimeStep(int offset, int delta) const
+{
+    int pdt = hourInWeek_ + offset;
+    const int nbTimeSteps = NombreDePasDeTempsPourUneOptimisation_;
+
+    if (const bool shifted_timestep = offset != 0; shifted_timestep)
```
Comment (flomnes) [inline src/solver/optimisation/variables/VariableManagement.cpp]:
```suggestion
    if (offset != 0)
```

File: src/solver/optimisation/variables/VariableManagement.cpp [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+ CorrespondanceVarNativesVarOptim_(CorrespondanceVarNativesVarOptim),
+ NumeroDeVariableStockFinal_(NumeroDeVariableStockFinal),
+ NumeroDeVariableDeTrancheDeStock_(NumeroDeVariableDeTrancheDeStock),
+ NombreDePasDeTempsPourUneOptimisation_(NombreDePasDeTempsPourUneOptimisation)
+{
+}
+
+int VariableManager::GetShiftedTimeStep(int offset, int delta) const
+{
+    int pdt = hourInWeek_ + offset;
```
Comment (flomnes) [inline src/solver/optimisation/variables/VariableManagement.cpp]: Pass `hourInWeek_` as an argument ? It's a bit hidden as it is

File: src/solver/optimisation/variables/VariableManagement.cpp [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    return CorrespondanceVarNativesVarOptim_[pdt].NumeroDeVariableDuPalierThermique[index];
+}
+
+int& VariableManager::NumberOfDispatchableUnits(unsigned int index,
+                                                unsigned int hourInWeek,
+                                                int offset,
+                                                int delta)
+{
+    hourInWeek_ = hourInWeek;
+    auto pdt = GetShiftedTimeStep(offset, delta);
```
Comment (flomnes) [inline src/solver/optimisation/variables/VariableManagement.cpp]: Every time we call `GetShiftedTimeStep`, we must set `hourInWeek_` first. It looks like `hourInWeek_` is an argument to `GetShiftedTimeStep`, right ?

**Review by flomnes — APPROVED** [review]
Minor changes, then you can change
- Pass hourInWeek to `GetShiftedTimeStep`, it's a hidden argument
- Remove temporary `shifted_timestep` (unused)

**Pull request merged**